hydromt/models/model_vector.py

```diff
@@ -0,0 +1,433 @@
+            gdf = gpd.read_file(join(self.root, fn_geom))
+            # geom + netcdf data
+            if fn is not None:
+                ds = GeoDataset.from_gdf(gdf, data_vars=ds)
+            # geom only
+            else:
+                ds = GeoDataset.from_gdf(gdf, keep_cols=True, cols_as_data_vars=True)
+        # netcdf only
+        elif fn is not None:
+            ds = GeoDataset.from_netcdf(ds)
+        else:
+            self.logger.info("No vector data found, skip reading.")
+            return
+
+        self.set_vector(ds)
+
+    def write_vector(
+        self,
+        fn: str = "vector/vector.nc",
+        fn_geom: str = "vector/vector.geojson",
+        ogr_compliant: bool = False,
+        **kwargs,
+    ):
+        """Write model vector to combined netcdf and geojson files.
+
+        Files are written at <root>/<fn> and at <root>/<fn_geom> respectively.
+
+        Three options are possible:
+
+            * The netcdf file contains the attribute data and the geojson file the
+                geometry vector data. Key-word arguments are passed to
+                :py:meth:`~hydromt.models.Model.write_nc`
+
+            * The netcdf file contains both the attribute and the geometry data
+                (fn_geom set to None). Key-word arguments are passed to
+                :py:meth:`~hydromt.models.Model.write_nc`
+
+            * The geojson file contains both the attribute and the geometry data
+                (fn set to None). This option is possible only if all data variables
+                are 1D. Else the data will be written to netcdf. Key-word arguments are
+                passed to :py:meth:`~hydromt.vector.GeoDataset.to_gdf`
+
+        Parameters
+        ----------
+        fn : str, optional
+            netcdf filename relative to model root,
+            by default 'vector/vector.nc'
+        fn_geom : str, optional
+            geojson filename relative to model root,
+            by default 'vector/vector.geojson'
+        ogr_compliant : bool
+            If fn only, write the netCDF4 file in an ogr compliant format
+            This makes it readable as a vector file in e.g. QGIS
+            see :py:meth:`~hydromt.vector.GeoBase.ogr_compliant` for more details.
+        **kwargs:
+            Additional keyword arguments that are passed to the `write_nc`
+            function.
+        """
+        if len(self.vector) == 0:
+            self.logger.debug("No vector data found, skip writing.")
+            return
+        self._assert_write_mode()
+        ds = self.vector
+
+        # If fn is None check that vector contains only 1D data
+        if fn is None:
+            # If the user did specify a reducer, data can be more than 1D data
+            if "reducer" in kwargs and kwargs["reducer"] is not None:
+                self.logger.warning(
+                    "If 2D/3D data found,"
+                    f"they will be reduced to 1D using {kwargs['reducer']}"
+                )
```

Comment:
Why is this functionality "hidden" in the kwargs? I would suggest to either make the reducer explicit or (preferred) remove it as I don't think you commonly want to reduce model input data at this stage.